client/src/app/shared/models/motions/workflow.ts

```diff
@@ -36,32 +36,7 @@ export class Workflow extends BaseModel<Workflow> {
      * Sorts the states of this workflow
      */
     public sortStates(): void {
-        // The first state has to be the first in a list later
-        let firstState: WorkflowState;
-        // the middle states are those with next states
-        const middleStates = [];
-        // the end states are those without any next states
-        const endStates = [];
-
-        for (let i = 0; i < this.states.length; i++) {
-            const state = this.states[i];
-            if (state.id === this.first_state_id) {
-                firstState = this.states[i];
-            } else {
-                if (state.next_states_id.length > 0) {
-                    middleStates.push(state);
-                } else {
-                    endStates.push(state);
-                }
-            }
-        }
-
-        // Do the actual sorting. Should (could) be handled later over better functions after PR 4411
-        middleStates.sort((a, b) => a.compare(b));
-        endStates.sort((a, b) => a.compare(b));
-
-        const sortedStates = [firstState].concat(middleStates).concat(endStates);
-        this.states = sortedStates;
+        this.states.sort((a, b) => a.compare(b));
```

Comment:
is `compare()` used somewhere else? If note, please remove this function and put the logic here...

Reply:
`compare` is dedicated model logic. Models should (actually always) be able to compare themselves with other models. We will probably need that for better sorting everywhere.
Since IDs are entirely virtual I can take it out of there for now.